libkbfs/ongoing_work_limiter.go

```diff
@@ -0,0 +1,40 @@
+package libkbfs
+
+import "sync"
+
+// OngoingWorkLimiter limits maximum number of routines that can work on a same
+// type of thing at the same time. For example, it can be used for limiting
+// number of ongoing rekeys.
+type OngoingWorkLimiter struct {
+	cond     sync.Cond
+	capacity int64
+}
+
+// NewOngoingWorkLimiter creates a new *OngoingWorkLimiter with capacity of
+// maxNumOngoingWorks.
+func NewOngoingWorkLimiter(maxNumOngoingWorks int) *OngoingWorkLimiter {
+	return &OngoingWorkLimiter{
+		capacity: int64(maxNumOngoingWorks),
+		cond:     sync.Cond{L: new(sync.Mutex)},
+	}
+}
+
+// WaitToStart blocks until the limiter would allow one more routine to start
+// working on the thing.
+func (owl *OngoingWorkLimiter) WaitToStart() {
+	owl.cond.L.Lock()
+	for owl.capacity <= 0 {
+		owl.cond.Wait()
```

Comment:
It would be nice if there was an option to wait for a `ctx.Done()` here as well.  Rekey task contexts can be canceled, right?  (It would have to fork a goroutine for the cond wait though, of course.)